src/cascadia/LocalTests_TerminalApp/SettingsTests.cpp

```diff
@@ -967,4 +969,160 @@ namespace TerminalAppLocalTests
         VERIFY_ARE_EQUAL(L"Ubuntu", settings._profiles.at(2)._name);
         VERIFY_ARE_EQUAL(L"Windows PowerShell", settings._profiles.at(3)._name);
     }
+
+    void SettingsTests::TestLayeringNameOnlyProfiles()
+    {
+        // This is a test discovered during GH#2782. When we add a name-only
+        // profile, it should only layer with other name-only profiles with the
+        // _same name_
+
+        const std::string settings0String{ R"(
+        {
+            "defaultProfile" : "{00000000-0000-5f56-a8ff-afceeeaa6101}",
+            "profiles": [
+                {
+                    "guid" : "{00000000-0000-5f56-a8ff-afceeeaa6101}",
+                    "name" : "ThisProfileIsGood"
+
+                },
+                {
+                    "name" : "ThisProfileShouldNotLayer"
+                },
+                {
+                    "name" : "NeitherShouldThisOne"
+                }
+            ]
+        })" };
+
+        const auto settings0Json = VerifyParseSucceeded(settings0String);
+
+        CascadiaSettings settings;

```

Comment:
does this test need to parse the defaults and validate them? it seems extraneous, but idk